task_processing/plugins/kubernetes/utils.py

```diff
@@ -190,6 +192,49 @@ def get_pod_volumes(volumes: PVector['DockerVolume']) -> List[V1Volume]:
     ]
 
 
+def get_kubernetes_empty_volume_mounts(
+    empty_volumes: PVector['EmptyVolume']
+) -> List[V1VolumeMount]:
+    """
+    Given a list of empty volume mounts, return a list corresponding to
+    the Kubernetes objects representing these mounts.
+    """
+    return [
+        V1VolumeMount(
+            mount_path=volume["container_path"],
+            name=get_sanitised_volume_name(
+                f"empty--{volume['container_path']}",
+                length_limit=63
+            ),
+        )
+        for volume in empty_volumes
+    ]
+
+
+def get_pod_empty_volumes(
+    empty_volumes: PVector['EmptyVolume']
+) -> List[V1Volume]:
+    """
+    Given a list of empty volume mounts, return a list corresponding to
+    the Kubernetes objects needed to tie the mounts to a Pod.
+    """
+    unique_volumes: Dict[str, 'EmptyVolume'] = {
+        get_sanitised_volume_name(f"empty--{volume['container_path']}", length_limit=63): volume
+        for volume in empty_volumes
+    }
+
+    return [
+        V1Volume(
+            name=name,
+            empty_dir=V1EmptyDirVolumeSource(
+                medium=volume.get('medium', ""),
```

Comment:
shouldn't the default here be None?

Reply:
The default is &quot;&quot; which means to use the node's default medium. Must be an empty string (default) or Memory.

https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1EmptyDirVolumeSource.md

that's how i interpreted this one.

Reply:
ah cool!